### PR TITLE
Adding windows support for caffe

### PR DESCRIPTION
**Pull request by hughed2**

### CODE REVIEW

**Comment by hughed2** [comment]
The way we link libraries for caffe into Scallop-TK needs to be fixed up, but this will work to push out the door in the short term and shouldn't affect non-Scallop-TK and non-Windows builds. An issue is that it relies on ntdll.lib, which I can supply but I don't believe comes standard on Windows and is a pain in the butt to get (it's part of WDK, which requires a larger install and can screw other settings).

File: CMakeLists.txt [hunk 1]
```diff
@@ -120,10 +120,10 @@ if( VIAME_ENABLE_CUDNN AND NOT VIAME_ENABLE_CUDA )
...
 endif()
 
-if( WIN32 AND VIAME_ENABLE_CAFFE )
-  message( FATAL_ERROR "Internal Caffe build on windows not currently supported, \
-    please build Caffe externally using the Caffe windows branch, if desired." )
-endif()
+#if( WIN32 AND VIAME_ENABLE_CAFFE )
+#  message( FATAL_ERROR "Internal Caffe build on windows not currently supported, \
+#    please build Caffe externally using the Caffe windows branch, if desired." )
+#endif()
```
Comment (linus-sherrill) [inline CMakeLists.txt]: If this is just dead code, could it be removed?

Comment (hughed2) [inline CMakeLists.txt]: Yes it could.

**Comment by hughed2** [comment]
Pending some passing tests, I've got a better fix coming in for the linking situation. I'll leave the script as is tonight and leave the PR open in case Matt wants to use this tomorrow morning, but linking gods willing there will be a new version up fairly soon.

**Comment by mattdawkins** [comment]
I've merged this into master through other means

**Pull request closed without merging**